elasticsearch/src/main/java/org/hibernate/search/elasticsearch/analyzer/impl/ScopedElasticsearchAnalyzerReference.java

```diff
@@ -6,26 +6,161 @@
  */
 package org.hibernate.search.elasticsearch.analyzer.impl;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import org.hibernate.search.analyzer.spi.AnalyzerReference;
 import org.hibernate.search.analyzer.spi.ScopedAnalyzerReference;
+import org.hibernate.search.exception.AssertionFailure;
+import org.hibernate.search.util.logging.impl.Log;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
 
 /**
  * @author Yoann Rodiere
  */
 public class ScopedElasticsearchAnalyzerReference extends ElasticsearchAnalyzerReference implements ScopedAnalyzerReference {
 
+	private static final Log LOG = LoggerFactory.make();
+
+	private ScopedElasticsearchAnalyzer analyzer;
+
+	private DeferredInitializationBuilder builder;
+
 	public ScopedElasticsearchAnalyzerReference(ScopedElasticsearchAnalyzer analyzer) {
-		super( analyzer );
+		this.builder = null;
+		this.analyzer = analyzer; // Already initialized
+	}
+
+	private ScopedElasticsearchAnalyzerReference(DeferredInitializationBuilder builder) {
+		this.builder = builder;
+		this.analyzer = null; // Not initialized yet
 	}
 
 	@Override
 	public ScopedElasticsearchAnalyzer getAnalyzer() {
-		return (ScopedElasticsearchAnalyzer) super.getAnalyzer();
+		if ( analyzer == null ) {
+			throw LOG.lazyRemoteAnalyzerReferenceNotInitialized( this );
+		}
+		return analyzer;
+	}
+
+	public boolean isInitialized() {
+		return analyzer != null;
+	}
+
+	public void initialize() {
+		if ( this.analyzer != null ) {
+			throw new AssertionFailure( "An Elasticsearch analyzer reference has been initialized more than once:" + this );
```

Comment:
A space after the dot would be nice.

(Yes, this is the one for which I expect crazy eyes)